libmamba/src/core/channel.cpp

```diff
@@ -231,19 +223,14 @@ namespace mamba
                 // that already contains the platform
                 else
                 {
-                    std::string cleaned_url = "", platform = "";
-                    util::split_platform(
-                        KNOWN_PLATFORMS,
-                        value,
-                        Context::instance().platform,
-                        cleaned_url,
-                        platform
-                    );
-                    if (!platform.empty())
+                    auto all_platforms = KNOWN_PLATFORMS;
```

Comment:
IIUC this creates a copy of `KNOWN_PLATFORMS` with the context channel added. Is that correct? I don't want to modify `KNOWN_PLATFORMS`.

Reply:
Indeed, this creates a copy. If you keep `all_platforms` and `context_platform` argument in the `split_platform` API, you don't need to do this.